src/main/java/org/openmaptiles/layers/WaterName.java

```diff
@@ -176,20 +189,32 @@ public void process(Tables.OsmWaterPolygon element, FeatureCollector features) {
         Geometry centerlineGeometry = lakeCenterlines.get(element.source().id());
         FeatureCollector.Feature feature;
         int minzoom = 9;
+        String place = element.place();
+        String clazz;
+        if ("bay".equals(element.natural())) {
+          clazz = FieldValues.CLASS_BAY;
+        } else if ("sea".equals(place)) {
+          clazz = FieldValues.CLASS_SEA;
+        } else {
+          clazz = FieldValues.CLASS_LAKE;
+          minzoom = 3;
+        }
         if (centerlineGeometry != null) {
           // prefer lake centerline if it exists
           feature = features.geometry(LAYER_NAME, centerlineGeometry)
```

Comment:
Based on this logic, do we always want to set minZoom = 3 when lake centerline is present?

```diff
FROM osm_water_lakeline
  WHERE geometry && bbox
-   AND ((zoom_level BETWEEN 9 AND 13 AND LineLabel(zoom_level, NULLIF(name, ''), geometry))
+   AND ((zoom_level BETWEEN 3 AND 13 AND LineLabel(zoom_level, NULLIF(name, ''), geometry))
      OR (zoom_level >= 14))
  UNION ALL
```

https://github.com/openmaptiles/openmaptiles/compare/v3.14..master#diff-201daa1c61c99073fe3280d440c9feca5ed2236b251ad454caa14cc203f952d1R32-R36

Reply:
OK this might end up behaving exactly the same way as setting minzoom=3 if centerline is present because the class should always "lake."  So as long as there aren't any centerlines in the dataset for "bay" or "sea" then it should be the same.